### DISCUSSION TITLE
How to override a standard action view (MVC action) from the UI

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#14543] [Q&A]
I want to be able to override an MVC-action view from the UI using Liquid. Typically, we use shape type and shape alternative to change the shape. But, how can we do it with standard MVC views?

For example, I want to change how the `Forbidden.cshtml` view looks like for one tenant. This view is part of the `OrchardCore.Diagnostics` module 

I can override these views by the following view in my custom theme `ThemeName/Views/OrchardCore.Diagnostics/Forbidden.cshtml`. But, how can I change that from the UI? Is there a way to to override the `NotFound` view from the `OrchardCore.Diagnostics` without having to change how that controller return? Meaning changing 

https://github.com/OrchardCMS/OrchardCore/blob/e228e452311fe16e74eaddcee1376c8b9662df8f/src/OrchardCore.Modules/OrchardCore.Diagnostics/Controllers/DiagnosticsController.cs#L17-L24
to something like this instead

```
return await httpStatusCode switch
{
    HttpStatusCode.Forbidden => New.Forbidden(),
    HttpStatusCode.NotFound => New.NotFound(),
    HttpStatusCode.BadRequest => New.BadRequest(),
    HttpStatusCode.Unauthorized => New.Unauthorized(),
    _ => New.Error(),
};
```

@jtkech you created the Diagnostic module, maybe you can shed light about how to override these MVC views?

**Comment by MikeAlhayek** [comment]
Guessing without having a custom theme, we currently can't override these views using the UI. I submitted a PR addressing this #14545

**Reply by jtkech** [reply]
We can override mvc view from the app that acts as a super theme, but would be the same for all tenants.

> using the UI

Do you mean the Template feature?

**Reply by MikeAlhayek** [reply]
@jtkech yes. check out the #14545 PR
